### PR TITLE
Copyonwrite/248/add devcontainer setup

### PR DESCRIPTION
**Pull request by copyonwrite**
## Description

This PR adds support for the [devcontainer](https://containers.dev/) standard.

devcontainers work OOTB for VSCode with the remote container extension and all versions of IDEA IDEs.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [x] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Closes # (248)

## Added tests?

- [ ] 👍 yes
- [x ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🤝 Documentation pages updated
- [x ] 🙅 no documentation needed
- [ ] (if applicable) generated OpenAPI docs for CRD changes

## Checklist

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [ x] My changes generate no new warnings
- [ x] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by MR2011** [comment]
I think we should add a small note to our Readme that the devcontainer can be used for development

**Review by MR2011 — APPROVED** [review]

**Comment by drochow** [comment]
Doesn't work on my system: 

image

🤔

**Comment by copyonwrite** [comment]
@drochow 

The error message complains about a conflicting port. Did you shutdown all other docker containers running? You are running on macOS? @MR2011  Had trouble with a conflicting setting in his VS Code preferences.

Do you try to use the Makefile to run stuff?

The whole idea is, that you just run/need VS Code locally with only the remote container plugin, everything else is provided by the devcontainer, key ready.

We can also look into this together, although I have no idea about macOS etc.

**Comment by drochow** [comment]
@copyonwrite 

Well, it's not complaining about other services running that would occupy the port. Nonetheless, I have shut down the other Services and getting this error.

> Do you try to use the Makefile to run stuff?
No I don't

File: README.md [hunk 1]
```diff
@@ -87,12 +87,37 @@ docker-compose --profile db --profile heureka up
...
 ### Makefile
 
-Alternatively, the application can be started by using the provided Makefile:
+The application can be started by using the provided Makefile:
 
 ```
 make start-all-heureka
 ```
 
+### Devcontainers
```
[the remark this answers is not in the document]
Comment (drochow) [reply README.md:96]: Also:

If we decide not to investigate the Intellij/OSX  combination, we may add to the documentation on which systems this setup is recommended/working.

**Comment by drochow** [comment]
@copyonwrite  please use commit signing to sign your commits

**Comment by drochow** [comment]
@copyonwrite Please use [conventional commit ](https://www.conventionalcommits.org/en/v1.0.0/) format for commit messages and (more important) pull request titles

**Review by drochow — APPROVED** [review]

**Review by MR2011 — APPROVED** [review]

**Pull request merged**